osu.Game/Skinning/DefaultSkin.cs

```diff
@@ -31,12 +33,23 @@ public DefaultSkin(IStorageResourceProvider resources)
         public DefaultSkin(SkinInfo skin, IStorageResourceProvider resources)
             : base(skin, resources)
         {
+            this.resources = resources;
             Configuration = new DefaultSkinConfiguration();
         }
 
         public override Texture GetTexture(string componentName, WrapMode wrapModeS, WrapMode wrapModeT) => null;
 
-        public override ISample GetSample(ISampleInfo sampleInfo) => null;
+        public override ISample GetSample(ISampleInfo sampleInfo)
+        {
+            foreach (var lookup in sampleInfo.LookupNames)
```

Comment:
Do legacy skins fall back correctly? I don't believe they fallback at all right now (just beatmap -> user skin -> null)?

If not, we can't really merge this until fallback to the legacy classic or at least this default skin is implemented.

Reply:
Do you mean the case where you have a legacy skin selected but it's not providing samples?

Reply:
That's it, yeah

Reply:
Will test tomorrow, not sure. But I can't immediately see why this would fail.

Reply:
Good catch, this is definitely broken. I'll look into this as a top priority because it's obvious this current skinning setup is too complex to follow (transformers / providing containers / `ISkinSource` fallbacks going on inside all of that).

Reply:
Falls back to the default skin, yup. I think it's fine for now.